gravitee-node-cache/gravitee-node-cache-plugin-redis/src/main/java/io/gravitee/node/plugin/cache/redis/configuration/RedisOptionsFactory.java

```diff
@@ -0,0 +1,219 @@
+import io.vertx.core.net.PemKeyCertOptions;
+import io.vertx.core.net.PemTrustOptions;
+import io.vertx.core.net.PfxOptions;
+import io.vertx.redis.client.RedisClientType;
+import io.vertx.redis.client.RedisOptions;
+import io.vertx.redis.client.RedisRole;
+import java.util.List;
+import java.util.Objects;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.util.StringUtils;
+
+/**
+ * @author Eric LELEU (eric.leleu at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+@Slf4j
+public class RedisOptionsFactory {
+
+    private static final String STORE_FORMAT_JKS = "JKS";
+    private static final String STORE_FORMAT_PEM = "PEM";
+    private static final String STORE_FORMAT_PKCS12 = "PKCS12";
+
+    public static RedisOptions build(RedisConfiguration configuration) {
+        return buildRedisOptions(configuration);
+    }
+
+    private static RedisOptions buildRedisOptions(RedisConfiguration configuration) {
+        final RedisOptions options = new RedisOptions();
+
+        final var sentinelConfiguration = configuration.getSentinelConfiguration();
+        if (sentinelConfiguration.isEnabled()) {
+            // Sentinels + Redis master / replicas
+            log.debug("Redis repository configured to use Sentinel connection");
+
+            options.setType(RedisClientType.SENTINEL);
+            List<HostAndPort> sentinelNodes = sentinelConfiguration.getNodes();
+            sentinelNodes.forEach(hostAndPort -> options.addConnectionString(hostAndPort.toConnectionString()));
+
+            if (!StringUtils.hasText(sentinelConfiguration.getMaster())) {
+                throw new IllegalStateException("Incorrect Sentinel configuration : parameter 'master' is mandatory!");
+            }
+            options.setMasterName(sentinelConfiguration.getMaster()).setRole(RedisRole.MASTER);
+            options.setPassword(sentinelConfiguration.getPassword());
+        } else {
+            // Standalone Redis
+            log.debug("Redis repository configured to use standalone connection");
+
+            options.setType(RedisClientType.STANDALONE);
+            options.setConnectionString(configuration.getHostAndPort().toConnectionString());
+        }
+
+        // SSL
+        if (configuration.getHostAndPort().isUseSsl()) {
+            log.debug("Redis repository configured with ssl enabled");
+
+            SslConfiguration sslConfiguration = configuration.getSslConfiguration();
+            options.getNetClientOptions().setSsl(true);
+            options.getNetClientOptions().setTrustAll(sslConfiguration.isTrustAll());
+
+            if ("NONE".equals(sslConfiguration.getHostnameVerificationAlgorithm())) {
+                options.getNetClientOptions().setHostnameVerificationAlgorithm("");
+            } else {
+                options.getNetClientOptions().setHostnameVerificationAlgorithm(sslConfiguration.getHostnameVerificationAlgorithm());
+            }
+
+            // TLS Protocols
+            options
```

Comment:
have a look at gravitee-node-vertx/src/main/java/io/gravitee/node/vertx/client/tcp, you find stuff that is already coded and tested for you. RedisClient uses a NetClient so a lot of what is in there shall be reused.